addons/l10n_pk_edi/models/account_move.py

```diff
@@ -0,0 +1,494 @@
+            "TaxCharged": self._l10n_pk_round_value(sales_tax_total),
+            "TotalAmount": self._l10n_pk_round_value(
+                sale_value + sales_tax_total + non_sales_tax_total
+            ),
+            "InvoiceType": 1,
+        }
+        if reference_number := self._get_l10n_pk_edi_fbr_ref_number():
+            line_json["RefUSIN"] = reference_number
+        return line_json
+
+    def _l10n_pk_edi_generate_invoice_json(self):
+        buyer = self.partner_id
+        lines = self.invoice_line_ids.filtered(lambda line: line.display_type not in (
+            'line_note', 'line_section', 'rounding'
+        ))
+        global_discount_line = lines.filtered(self._l10n_pk_is_global_discount)
+        lines -= global_discount_line
+        tax_details = self._prepare_invoice_aggregated_taxes()
+        tax_details_per_record = tax_details.get("tax_details_per_record")
+        total_quantity = sum(line.quantity for line in lines)
+        # Calculate the total discount as the sum of the global discount and the line discounts
+        total_discount = (
+            sum(line.balance for line in global_discount_line) * (1 if self.is_inbound() else -1)
+        ) + sum((line.price_unit * line.quantity) * (line.discount / 100) for line in lines)
+        # Total Sale Value (Excluding Discount and Tax Amount)
+        total_sale_value = sum(line.price_unit * line.quantity for line in lines)
+
+        json_payload = {
+            "InvoiceNumber": "",
+            "POSID": self._get_l10n_pk_edi_partner_pos_id(),
+            "USIN": self.name.replace('/', '_'),
+            "DateTime": self.format_timestamp(self.invoice_date),
+            "BuyerNTN": buyer.vat or "",
+            "BuyerCNIC": None,
+            "BuyerName": buyer.name or "",
+            "BuyerPhoneNumber": None,
+            "TotalBillAmount": self._l10n_pk_round_value(
+                tax_details.get("base_amount") + tax_details.get("tax_amount")
+            ),
+            "TotalQuantity": total_quantity,
+            "TotalSaleValue": self._l10n_pk_round_value(total_sale_value),
+            "TotalTaxCharged": self._l10n_pk_round_value(tax_details.get("tax_amount")),
+            "Discount": self._l10n_pk_round_value(total_discount),
+            "FurtherTax": 0.00,
+            "PaymentMode": self._get_l10n_pk_edi_payment_mode(),
+            "RefUSIN": None,
+            "InvoiceType": self._get_l10n_pk_edi_invoice_type(),
+            "Items": [
+                self._get_l10n_pk_edi_line_details(index, line, tax_details_per_record.get(line, {}))
+                for index, line in enumerate(lines, start=1)
+            ],
+        }
+        return json_payload
+
+    def _get_l10n_pk_total_tax_values(self, tax_lines):
+        sales_tax_total = 0.0
+        non_sales_tax_total = 0.0
+        sale_group_ids = (
+            self._get_l10n_pk_sales_groups()
+            + self._get_l10n_pk_withhold_groups()
+        ).ids
```

Comment:
```suggestion
            self._get_l10n_pk_sales_groups().ids
            + self._get_l10n_pk_withhold_groups().ids
        )
```
Concatenation of list?